packages/vitest/src/integrations/chai/poll.ts

```diff
@@ -90,9 +80,24 @@ export function createExpectPoll(expect: ExpectStatic): ExpectStatic['poll'] {
               }
               catch (err) {
                 lastError = err
-                intervalId = setTimeout(check, interval)
+                if (!chai.util.flag(assertion, '_isLastPollAttempt')) {
+                  intervalId = setTimeout(check, interval)
+                }
               }
             }
+            timeoutId = setTimeout(async () => {
+              clearTimeout(intervalId)
+              chai.util.flag(assertion, '_isLastPollAttempt', true)
+              await check()
```

Comment:
this check can succeed or throw an error, this needs special handling

Reply:
hey

from what i can see the `check` function declaration is already using a try/catch block and handling the error.

if the assertion succeeds, it will resolve the promise successfuly
if the assertion fails, it will end in the catch block and wont throw an error

so the only scenario check can throw an error is if `chai.util.flag` access for some reason throw. if you want me to handle that case, can you provide more details on how? you want specific behavior or just logging?

im adding test cases for both scenarios: sucess/failure on last attempt

i added two new test cases to cover such scenarios

Reply:
I see, I guess the name is a bit misleading. We don't need an error handling here then